server/channels/app/support_packet.go

```diff
@@ -19,61 +15,86 @@ import (
 	"github.com/mattermost/mattermost/server/public/shared/request"
-	"github.com/mattermost/mattermost/server/v8/config"
-)
-
-const (
-	cpuProfileDuration = 5 * time.Second
 )
 
 func (a *App) GenerateSupportPacket(c request.CTX, options *model.SupportPacketOptions) []model.FileData {
-	// If any errors we come across within this function, we will log it in a warning.txt file so that we know why certain files did not get produced if any
-	var warnings []string
-
-	// Creating an array of files that we are going to be adding to our zip file
-	fileDatas := []model.FileData{}
-
 	// A array of the functions that we can iterate through since they all have the same return value
 	functions := map[string]func(c request.CTX) (*model.FileData, error){
-		"support package": a.generateSupportPacketYaml,
-		"plugins":         a.createPluginsFile,
-		"config":          a.createSanitizedConfigFile,
-		"cpu profile":     a.createCPUProfile,
-		"heap profile":    a.createHeapProfile,
-		"goroutines":      a.createGoroutineProfile,
-		"metadata":        a.createSupportPacketMetadata,
+		"support packet": a.generateSupportPacketYaml,
+		"plugins":        a.createPluginsFile,
+		"config":         a.createSanitizedConfigFile,
+		"cpu profile":    a.createCPUProfile,
+		"heap profile":   a.createHeapProfile,
+		"goroutines":     a.createGoroutineProfile,
+		"metadata":       a.createSupportPacketMetadata,
 	}
 
 	if options.IncludeLogs {
 		functions["mattermost log"] = a.GetMattermostLog
 		functions["notification log"] = a.getNotificationsLog
 	}
 
-	for name, fn := range functions {
-		fileData, err := fn(c)
-		if err != nil {
-			c.Logger().Error("Failed to generate file for support package", mlog.Err(err), mlog.String("file", name))
-			warnings = append(warnings, err.Error())
-		}
+	// If any errors we come across within this function, we will log it in a warning.txt file so that we know why certain files did not get produced if any
+	var warnings *multierror.Error
+	// Creating an array of files that we are going to be adding to our zip file
+	var fileDatas []model.FileData
+	var wg sync.WaitGroup
+	var mut sync.Mutex // Protects warnings and fileDatas
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
 
-		if fileData != nil {
-			fileDatas = append(fileDatas, *fileData)
+		for name, fn := range functions {
+			fileData, err := fn(c)
+			mut.Lock()
+			if err != nil {
+				c.Logger().Error("Failed to generate file for Support Packet", mlog.String("file", name), mlog.Err(err))
+				warnings = multierror.Append(warnings, err)
+			}
+
+			if fileData != nil {
+				fileDatas = append(fileDatas, *fileData)
+			}
+			mut.Unlock()
 		}
+	}()
+
+	// Run the cluster generation in a separate goroutine as CPU profile generation and file upload can take a long time
+	if cluster := a.Cluster(); cluster != nil && *a.Config().ClusterSettings.Enable {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			files, err := cluster.GenerateSupportPacket(c, options)
```

Comment:
This will trigger plugins to generate multiple support packets over and over again, for cluster, those should only generated once right?

Reply:
Nevermind, I think this is being handled in the gossip client.

Reply:
Yes, the `generateSupportPacketHandler` in `GossipClient` does not invoke plugins.